homeassistant/components/xiaomi_miio/select.py

```diff
@@ -0,0 +1,253 @@
+    KEY_COORDINATOR,
+    KEY_DEVICE,
+    MODEL_AIRHUMIDIFIER_CA1,
+    MODEL_AIRHUMIDIFIER_CA4,
+    MODEL_AIRHUMIDIFIER_CB1,
+    MODELS_HUMIDIFIER,
+    SERVICE_SET_LED_BRIGHTNESS,
+)
+from .device import XiaomiCoordinatedMiioEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_LED_BRIGHTNESS = "led_brightness"
+
+SERVICE_TO_METHOD = {
+    SERVICE_SET_LED_BRIGHTNESS: {
+        "method": "async_set_led_brightness",
+        "property": ATTR_LED_BRIGHTNESS,
+    },
+}
+
+
+@dataclass
+class SelectorType:
+    """Class that holds device specific info for a xiaomi aqara or humidifier selectors."""
+
+    name: str = None
+    short_name: str = None
+    unit_of_measurement: str = None
+    icon: str = None
+    device_class: str = None
+    min: float = None
+    max: float = None
+    mode: str = None
+    options: list = None
+    step: float = None
+    service: str = None
+
+
+SELECTOR_TYPES = {
+    FEATURE_SET_LED_BRIGHTNESS: SelectorType(
+        name="Led brightness",
+        icon="mdi:brightness-6",
+        short_name=ATTR_LED_BRIGHTNESS,
+        options=["Bright", "Dim", "Off"],
+        service=SERVICE_SET_LED_BRIGHTNESS,
+    ),
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Selectors from a config entry."""
+    entities = []
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+        host = config_entry.data[CONF_HOST]
+        token = config_entry.data[CONF_TOKEN]
+        model = config_entry.data[CONF_MODEL]
+        device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+        coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+        device_features = 0
+        entity_class = None
+
+        _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+
+        if model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB
+            entity_class = XiaomiAirHumidifierSelector
+        elif model in [MODEL_AIRHUMIDIFIER_CA4]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA4
+            entity_class = XiaomiAirHumidifierMiotSelector
+        elif model in MODELS_HUMIDIFIER:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER
+            entity_class = XiaomiAirHumidifierSelector
+        else:
+            return
+
+        for feature in SELECTOR_TYPES:
+            selector = SELECTOR_TYPES[feature]
+            if feature & device_features and feature in SELECTOR_TYPES:
+                entities.append(
+                    entity_class(
+                        f"{config_entry.title} {selector.name}",
+                        device,
+                        config_entry,
+                        f"{selector.short_name}_{config_entry.unique_id}",
+                        selector,
+                        coordinator,
+                    )
+                )
+
+    async_add_entities(entities, update_before_add=False)
```

Comment:
Default value of `update_before_add` is `False`.
```suggestion
    async_add_entities(entities)
```